### PR TITLE
[#48] Added profile settings page UI with mock data

### LINKED ISSUE #48: Add a user settings page
**Is your feature request related to a problem? Please describe.**

This is a front end component, please refer to the Figma on the wiki for the current design. If there is no design, feel free to add your own to fit the criteria.

We need a page where the user can change their profile details. The route can be “/settings”

**Describe the solution you'd like**
- A header (see: customizable header component) with the page title: User Settings
- (The header also needs a save button passed in)
- Their profile banner and picture (clicking on them should open a dialog to select a file)
- Their username, nickname and bio (with some way to change them with a text input)

There is an example for adding a button as a child of customizable header on the issue #40 

Here is mock data for a user if needed 

```js
// mock-data.js
const sampleUser = {
    id: 1,
    username: "username",
    nickname: "nickname",
    profileBanner: "https://i.imgur.com/PcEvuMw.png",
    profilePic: "https://i.imgur.com/qZImY9j.jpg",
    followers: 11,
    following: 22,
    joinedDate: 1646627469012
}

export default sampleUser
```

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by bikkku**
# Description:
Please provide a summary of changes made and the relevant issue that has been worked on. Also list any major changes made to project structure/dependencies if applicable

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My code follows the code style of this project
- [ ] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: frontend/src/pages/profileSettings/ProfileSettingsController.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    user={user}
+    updateProfile={updateProfile}
+    avatarOpen = {avatarModalOpen}
+    handleAvatarOpen={handleAvatarOpen}
+    handleAvatarClose={handleAvatarClose}
+    />
+  );
+} 
+
+export default ProfileSettingsController
```
Comment (Billsong19) [inline frontend/src/pages/profileSettings/ProfileSettingsController.jsx:42]: You might want to add a new line at the end of the file as a consistency thing, even though we haven't established that as a convention/it passes eslint

File: frontend/src/pages/profileSettings/profilesettings.module.scss [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+.avatarEditor {
+    border-radius: 4px;
+    border-color: black;
+    border-width: 2px;
+}
+
+.upload {
+  margin-top: 8px;
+  margin-bottom: 8px;
+}
```
Comment (Billsong19) [inline frontend/src/pages/profileSettings/profilesettings.module.scss:96]: same here with the new line

**Review by Billsong19 — COMMENTED** [review]
LGTM, with some nitpicks, up to you to change

File: frontend/src/pages/profileSettings/ProfileSettingsController.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import MockUser from './mock-user'
+
+/**
+ * This page is where users can edit their profile
+ */
+const ProfileSettingsController = () => {
+  const [user, setUser] = useState([]);
+  const [avatarModalOpen, setAvatarModalOpen] = useState(false);
+
+  // const [bannerModalOpen, setBannerModalOpen] = useState(false);
```
Comment (seanhogunkim) [inline frontend/src/pages/profileSettings/ProfileSettingsController.jsx:12]: Remove unnecessary comment

Comment (bikkku) [reply frontend/src/pages/profileSettings/ProfileSettingsController.jsx:12]: it is necessary

File: frontend/src/pages/profileSettings/ProfileSettingsView.jsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            </div>
+        </div>
+        <p className={classes.subtext} style={{ marginBottom: '14px' }}>
+            {user.bio}
+        </p>
+
+        <Divider className={classes.divider} variant="middle" />
+
+        <div className={classes.save}>
+            <Button type="submit" variant="contained" onClick={updateProfile()}>
```
Comment (nzbasic) [inline frontend/src/pages/profileSettings/ProfileSettingsView.jsx]: updateProfile() will be called everytime the view is rendered, you should remove the parenthesis

**Review by seanhogunkim — APPROVED** [review]
LGTM after a couple small changes

File: frontend/src/pages/profileSettings/ProfileSettingsView.jsx (hunk 4 above)
Comment (bikkku) [reply frontend/src/pages/profileSettings/ProfileSettingsView.jsx]: done

**Review by Billsong19 — APPROVED** [review]

**Pull request merged**